test/Maps/MapReader.test.cpp

```diff
@@ -0,0 +1,24 @@
+#include "Maps/MapReader.h"
+#include "Maps/MapType.h"
+#include "gtest/gtest.h"
+#include <string>
+
+TEST(MapReader, EmptyFilename) {
+	EXPECT_THROW(MapReader::Read(std::string(), MapType::Map), std::runtime_error);
+	EXPECT_THROW(MapReader::Read(std::string(), MapType::Save), std::runtime_error);
+}
```

Comment:
This test would probably be more clear with a literal empty string `""`.

Reply:
Okay, I'll add them in.

Reply:
Slight note, but the `std::string` wrapping/conversion is automatic. There is an implicit conversion from C-style strings. The explicit wrapping also begs the question, why is this different from the filenames being passed in the code below?

Reply:
Well, I can remove the quotes so the c-style string conversion isn't happening again. I thought it was clear enough without the quotes anyways.